spyder/widgets/sourcecode/tests/test_extra_selections.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+"""Tests for editor extra selections (decorations)."""
+
+# Third party imports
+import pytest
+from qtpy.QtGui import QTextCursor, QTextFormat
+
+# Local imports
+from spyder.utils.qthelpers import qapplication
+from spyder.widgets.sourcecode.codeeditor import CodeEditor
+
+
+# --- Fixtures
+# -----------------------------------------------------------------------------
+def construct_editor(*args, **kwargs):
+    """Construct editor with some text for testing extra selections."""
+    app = qapplication()
+    editor = CodeEditor(parent=None)
+    kwargs['language'] = 'Python'
+    editor.setup_editor(*args, **kwargs)
+
+    text = ("def some_function():\n"
+            "    some_variable = 1\n"
+            "    some_variable += 2\n"
+            "    return some_variable\n"
+            "# %%")
+    editor.set_text(text)
+    return editor
+
+
+def test_extra_selections(qtbot):
+    """Test extra selections."""
+    editor = construct_editor()
+
+    # Move cursor over 'some_variable'
+    editor.go_to_line(2)
+    cursor = editor.textCursor()
+    cursor.movePosition(QTextCursor.Right, n=5)
+    editor.setTextCursor(cursor)
+
+    qtbot.waitUntil(lambda: len(editor.extraSelections()) >= 6, timeout=2000)
+    print(len(editor.extraSelections()))
```

Comment:
Is this print really needed?

Reply:
I think it ts a good practice to have prints inside tests cause when the test fails you get more debugging info. Dont know of it applies to this though.

Reply:
Hmm no, I was using it locally for know the variables when the test failed. I'll remove it

Reply:
>I was using it locally for know the variables when the test failed

Well.. if the test fails, it would be nice to have that information, no? otherwise we just get an assertion error that might not provide enough info

Reply:
This print wasn't giving too much information.

But adding print to test could be a good practice, for easy now what is failing in the CI when a test fails.